fix some npm audit issues

kevcube opened the pull request:
Issue:

## What I did

fix npm audit issues for trim (inside addon-docs), cpy (inside lib/core-server)

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this. For additional guidance: https://storybook.js.org/docs/react/contribute/how-to-contribute

Maintainers: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

-->

shilman reviewed (changes requested):
Hi @kevcube, thanks for submitting this PR!

I can't comment on the `cpy` changes without digging in, but I can say that the MDX part of the PR is a breaking change so we can't merge that part of this PR. Storybook already supports MDX2, however, and you can enable it like this:

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#opt-in-mdx2-support

yannbf commented:
> Hi @kevcube, thanks for submitting this PR!
> 
> I can't comment on the `cpy` changes without digging in, but I can say that the MDX part of the PR is a breaking change so we can't merge that part of this PR. Storybook already supports MDX2, however, and you can enable it like this:
> 
> https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#opt-in-mdx2-support

The changes seem good in the `cpy` front!

About the mdx change, this relates to this PR as well: https://github.com/storybookjs/storybook/pull/18203
If you'd like, you could remove the MDX changes from this PR and watch for updates in that other PR, then leave the cpy changes which seem good and at least will solve some of the audit issues

joshwooding committed:
Move memoizerific to dependencies in theming

joshwooding committed:
Move memoizerific and qs to dependencies in router

joshwooding committed:
Move memoizerific and qs to dependencies in ui

Andarist committed:
Fixed PnP compatibility for bundled ui and router packages

yannbf committed:
chore(CI): reenable cra pnp tests

yannbf committed:
fix: change the way presets are required

Andarist committed:
Merge branch 'next' into fix/pnp-compat-2

yannbf committed:
telemetry: strip out preset from addon name

also move getStorybookMetadata method to the bottom so it's used after the definition
of computeStorybookMetadata

yannbf committed:
Merge pull request #18412 from storybookjs/fix/pnp-compat-2

Core: fix PnP compatibility for @storybook/ui and @storybook/router packages

shilman committed:
Merge pull request #18442 from storybookjs/fix/telemetry-strip-out-preset

Telemetry: Strip out preset from addon name

joshwooding committed:
Warn when storyName usage is detected in CSF3 stories

shilman committed:
Merge pull request #18464 from joshwooding/fix-csf3-storyName-storyStoreV7

Story index: Warn on `storyName` in CSF3 exports

shilman committed:
Merge pull request #18367 from joshwooding/fix-process-is-not-defined

Commit:
Remove duplicate statement

`title: 'DocumentScreen',` appears twice in the component meta.

joshwooding committed:
Core: Fix process is not defined when using components

shilman committed:
Merge pull request #18469 from joshwooding/move-memoizerific-to-deps

Core: Fix process is not defined when using components

shilman committed:
Merge pull request #18467 from mattevenson/patch-1

[documentation] Remove duplicate statement in React snippet from docs

kevcube committed:
fix cpy vulnerabilities

kevcube committed:
bump to 9.0.1 instead of 9

kevcube commented:
@yannbf please re-review, I've excluded the mdx vulnerabilities and only touched `cpy` 😄
